src/raster/r.viewshed.exposure/r.viewshed.exposure.py

```diff
@@ -0,0 +1,1391 @@
+
+
+def clean_temp(pid):
+    """Remove temporary files of the current processes
+    :param pid: Process ID whos tempfiles to remove
+    :type pid: int
+    """
+    from pathlib import Path
+    from shutil import rmtree
+
+    tempfile = Path(grass.tempfile(create=False))
+    for path in tempfile.parent.glob(str(pid) + ".*"):
+        if path.is_file():
+            grassutils.try_rmdir(path)
+        else:
+            rmtree(path)
+
+
+def do_it_all(global_vars, target_pts_np):
+    """Conduct weighted and parametrised partial viewshed and cummulate it with
+    the previous partial viewsheds
+    :param target_pts_np: Array of target points in global coordinate system
+    :type target_pts_np: ndarray
+    :return: 2D array of weighted parametrised cummulative viewshed
+    :rtype: ndarray
+    """
+    # Set counter
+    counter = 1
+
+    # Get variables out of global_vars dictionary
+    reg = global_vars["region"]
+    exp_range = global_vars["range"]
+    flagstring = global_vars["flagstring"]
+    r_dsm = global_vars["r_dsm"]
+    v_elevation = global_vars["observer_elevation"]
+    refr_coeff = global_vars["refr_coeff"]
+    memory = global_vars["memory"]
+    parametrise_viewshed = global_vars["param_viewshed"]
+    dsm_type = global_vars["dsm_type"]
+    b_1 = global_vars["b_1"]
+    cores = global_vars["cores"]
+    tempname = global_vars["tempname"]
+
+    # Create empty viewshed
+    np_cum = np.empty((reg.rows, reg.cols), dtype=np.single)
+    np_cum[:] = np.nan
+    tmp_vs = "{}_{}".format(tempname, os.getpid())
+
+    for target_pnt in target_pts_np:
+
+        # Display a progress info message
+        grass.percent(counter, len(target_pts_np), 1)
+        grass.verbose(
+            "Processing point {i} ({p:.1%})".format(
+                i=int(target_pnt[0]), p=counter / len(target_pts_np)
+            )
+        )
+
+        # Global coordinates and attributes of target point T
+        t_glob = target_pnt[1:]
+
+        # ======================================================================
+        # 1. Set local computational region: +/- exp_range from target point
+        # ======================================================================
+        # ensure that local region doesn't exceed global region
+        loc_reg_n = min(t_glob[1] + exp_range + reg.nsres / 2, reg.north)
+        loc_reg_s = max(t_glob[1] - exp_range - reg.nsres / 2, reg.south)
+        loc_reg_e = min(t_glob[0] + exp_range + reg.ewres / 2, reg.east)
+        loc_reg_w = max(t_glob[0] - exp_range - reg.ewres / 2, reg.west)
+
+        # pygrass sets region for pygrass tasks
+        lreg = deepcopy(reg)
+        lreg.set_bbox(Bbox(loc_reg_n, loc_reg_s, loc_reg_e, loc_reg_w))
+        lreg.set_raster_region()
+
+        # Create processing environment with region information
+        c_env = os.environ.copy()
+        c_env["GRASS_REGION"] = grass.region_env(
+            n=loc_reg_n, s=loc_reg_s, e=loc_reg_e, w=loc_reg_w
+        )
```

Comment:
When I run the test, I am getting:
```
WARNING: East-west resolution does not equal north-south resolution. The
         viewshed computation assumes the cells are square, so in this case
         this may result in innacuracies.
```

I am wondering if that's not caused by some incorrect alignment, shouldn't you use:
```suggestion
        c_env["GRASS_REGION"] = grass.region_env(
            n=loc_reg_n, s=loc_reg_s, e=loc_reg_e, w=loc_reg_w, align=r_dsm
        )
```

Reply:
> When I run the test, I am getting:
> 
> ```
> WARNING: East-west resolution does not equal north-south resolution. The
>          viewshed computation assumes the cells are square, so in this case
>          this may result in innacuracies.
> ```
> 
> I am wondering if that's not caused by some incorrect alignment, shouldn't you use:

The alignment didn't work properly if sampling points were defined manually (i.e. if they were not exactly in the middle of a raster cell). Corrected in https://github.com/OSGeo/grass-addons/pull/708/commits/4ae225ed361d80c8ae7e4ccc0d5c8d2bff46e726